hack/update-codegen.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+# more info: https://cloud.redhat.com/blog/kubernetes-deep-dive-code-generation-customresources
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+mkdir -p ./api/clustergroupupgradesoperator
+cd ./api/clustergroupupgradesoperator
+ln -s ../v1alpha1 ./v1alpha1
```

Comment:
Is api/clustergroupupgradesoperator intended to be a working dir that doesn't already exist and shouldn't be committed? In the case of failure, the script is exiting before this is cleaned up at the end of the script, and this ln command, for example, fails.

Reply:
Yeah it's intended to be like this i.e used just to keep code-gen happy and not committing. But added trap for cleanup.